pkg/envinfo/envinfo.go

```diff
@@ -16,13 +16,20 @@ import (
 
 // ComponentSettings holds all component related information
 type ComponentSettings struct {
-	Name        string              `yaml:"Name,omitempty"`
+	Name string `yaml:"Name,omitempty"`
+
 	Namespace   string              `yaml:"Namespace,omitempty"`
 	URL         *[]EnvInfoURL       `yaml:"Url,omitempty"`
 	PushCommand *EnvInfoPushCommand `yaml:"PushCommand,omitempty"`
+	// AppName is the application name. Application is a virtual concept present in odo used
+	// for grouping of components. A namespace can contain multiple applications
+	AppName string `yaml:"AppName,omitempty" json:"AppName,omitempty"`
 
 	// DebugPort controls the port used by the pod to run the debugging agent on
 	DebugPort *int `yaml:"DebugPort,omitempty"`
+
+	// ComponentType stores the type of component - this information is available in devfile as well
+	ComponentType string `yaml:"ComponentType,omitempty"`
```

Comment:
Why are we storing it here again if it is already available in devfile?

Reply:
Because then the list would have to parse the devfile for just 1 field

Reply:
That is completely fine. Parsing devfile is not expensive. Definitely better than storing duplicated values.

Reply:
done, removed the ComponentType local envinfo